### PR TITLE
Improve python3 and docker support

### PR DESCRIPTION
**Pull request by route1rodent**
Contents of this PR:

- Fix setup on Python 3
- Simplify docker support and scripts
- Some additions and clarifications in the README (please review the changes so we find consensus on editing CSVs)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
-    name = 'Pokedex',
-    version = '0.1',
-    zip_safe = False,
-    packages = find_packages(),
-    package_data = {
+    name='Pokedex',
+    version='0.2',
+    zip_safe=False,
+    packages=find_packages(),
+    hiddenimports=['six'],
```
Comment (magical) [inline setup.py]: What is `hiddenimports`?

File: setup.py [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-import sys
-
 setup(
-    name = 'Pokedex',
-    version = '0.1',
-    zip_safe = False,
-    packages = find_packages(),
-    package_data = {
+    name='Pokedex',
+    version='0.2',
```
Comment (magical) [inline setup.py]: We've never bumped the version before. I don't see any reason to start now.

File: README.md [hunk 3]
```diff
@@ -1,31 +1,101 @@
...
-I started on an experiment with switching to YAML for data storage some time ago, for a variety of reasons.  It's finally starting to show some promise — all of gen 7 was dumped to a YAML format, then loaded into the database from there — but it'll take a lot more work to get this usable.  The intended _upsides_ are:
+The project needs to be modernized and to have a lot of rough edges fixed that would also make it easier
+to maintain for the games and generations to come.
 
-- The data will include everything from older games, so you don't have to guess!  Also, the site will handle older games correctly, probably!
-- Many more filtering and searching tools on veekun, since I won't have to fight SQL to write them!
-- More interesting data we've never had before, like trainer teams and overworld items!  And models?  Maps, even?  Who knows, but working on this stuff should be easier with all this existing code in place!
+### Status of the YAML migration
+
+**Update from 18.06.2017**
```
Comment (magical) [inline README.md]: 2017-06-18

File: docker-pokedex [hunk 4]
```diff
@@ -1,3 +0,0 @@
-#!/usr/bin/env bash
-
-docker-compose run --rm pokedex "$@"
```
Comment (magical) [inline docker-pokedex]: It looks like maybe you moved this to `bin/docker-pokedex` but forgot to `git add` the new file.

File: NOTES.md [hunk 5]
```diff
@@ -1,3 +1,7 @@
+# Notes
+
+## Item Questions
+
 QUESTIONS ABOUT ITEMS
```
Comment (magical) [inline NOTES.md]: Why rename this file? It's ancient (last touched 2011) and i have no idea if it's still relevant - could probably just delete it. Or leave it alone.

File: setup.py (hunk 1 above)
Comment (route1rodent) [reply setup.py]: without this python 3.7 was complaining with an error:
```
ImportError: The 'six' package is required; normally this is bundled with this package so if you get this warning, consult the packager of your distribution.
```

I used this solution as suggested here https://github.com/pyinstaller/pyinstaller/issues/1773

File: NOTES.md (hunk 5 above)
Comment (route1rodent) [reply NOTES.md]: it was for code organisation. but it's old, I guess we can remove it

File: setup.py (hunk 1 above)
Comment (magical) [reply setup.py]: That's fine. I just want to understand what it is and why it is necessary in this situation. I've never seen the `hiddenimports` directive before. There's no mention of it in the setuptools docs; the only documentation of it i could find is in the docs for Pyinstaller, which afaik we don't use. [This stackoverflow answer](https://stackoverflow.com/questions/15229658/pyinstaller-what-are-hiddenimports-and-hooks) implies that it's mainly needed for cases where an import name is constructed dynamically, e.g. with `__import__`, but we don't do that either.

Also, i'll note that it runs fine under Python 3.7 on [Travis](https://travis-ci.org/veekun/pokedex/jobs/656835696) and on my machine, so there's clearly more to the story here.

Comment (route1rodent) [reply setup.py]: I removed it. It's weird, it also works for me now... maybe I did something stupid before.

Comment (magical) [reply setup.py]: Oh weird. ¯\\\_(ツ)\_/¯

**Pull request merged**